glimr/search.py

```diff
@@ -120,6 +123,43 @@ def __init__(
         space["fit_kwargs"] = fit_kwargs
         space["loader"] = loader
         self._space = space
+        self.cv = cv
+
+        # if cv is 'True', check if search space contains cv parameters
+        if cv:
+            data_keys = list(space["data"].keys())
+            if "cv_fold_index" in data_keys and "cv_fold_index" in data_keys:
+                pass
+            else:
+                raise ValueError(
+                    "For cross validation, data dictionary should contain two `cv_fold_index`, `cv_folds` keys"
+                )
+
+        # check if data loader has two input aurguments `cv_fold_index`, `cv_folds`
+        if cv:
+            args = inspect.getfullargspec(loader)[0]
+            if "cv_fold_index" in args and "cv_folds" in args:
+                pass
+            else:
+                raise ValueError(
+                    """For cross validation, data loader must have two aurguments `cv_fold_index`, `cv_folds`.:
+                
+                Example:
```

Comment:
It's good to check the function signature for the required kwargs.

Probably better to keep the error message short. Put the rest of this in the documentation.